### ISSUE TITLE
Black screen on application launch

### ISSUE THREAD

**Issue by klaidliadon** [hrydgard/ppsspp#13273]
### What happens?
When opening the application, it shows a balck screen instead of the menu. Pressing the controls does nothing, only pressing the back button exits the app, and while exiting is possible to see the menu for a brief moment.

### What *should* happen?
The application should open normally

### What hardware, operating system, and PPSSPP version?  On desktop, GPU matters for graphical issues.
Sony Bravia KD49XH8096
Android TV with latest firmware (Android version 9)
PS4 DualShock connected

**Comment by klaidliadon** [comment]
I found this thread that suggested that Vulkan Backend support missing was the culprit.

https://forums.ppsspp.org/showthread.php?tid=20409

I changed the backend to OpenGL in the phone to see how the the file `/sdcard/PSP/SYSTEM/ppsspp.ini` changes.

```ini
[Graphics]
EnableCardboardVR = False
CardboardScreenSize = 50
CardboardXShift = 0
CardboardYShift = 0
ShowFPSCounter = 2
GraphicsBackend = 0 (OPENGL)
FailedGraphicsBackends = 
DisabledGraphicsBackends = 
```
After all changing the value of `GraphicsBackend` to `0` manually, by editing file with a text editor on the TV, now the app runs normally.

**Comment by aisanu** [comment]
I found this also happening on my Sony Bravia X8000H (its australian equivalent for X800H). My current workarounds was installing older version of PPSSPP that runs fine on the TV, and copy the ppsspp.ini file. I suspect this is caused by the same issue you faced.

**Comment by hrydgard** [comment]
This should not be closed - it's really bad if you need to edit the ini to make it work. It's possible that the device has a Broken vulkan driver or that ppsspp makes some mistake in detection.

**Comment by sum2012** [comment]
Can we blacklist some device to force to use opengl ?
Another way that we can use opengl as default.

**Comment by hrydgard** [comment]
yes, we could. We need to know the device name and model though as exposed by their Android implementations. This can be seen in PPSSPP system information as Name.

**Comment by unknownbrackets** [comment]
Currently we default to Vulkan on 8.1+, which includes this 9 device:
https://github.com/hrydgard/ppsspp/blob/d7fa5928d4f7ff7d8b62d4619c987aa1808f5007/Core/Config.cpp#L584-L585

An adb logcat might be helpful in case it's some detectable error.  We could force restart to Vulkan or trigger a Java prompt to do so.

-[Unknown]

**Comment by vittrup88** [comment]
I have the same issue.
Black screen but then clicking back on remote I can see menu in a brief moment.
My tv is Philips 50PUS8535 - and newest version of PPSSPP
I tried changing GraphicsBackend to 0 but does not help (also changed language to en_US).

the same problem when GraphicsBackend is set to 3 Vulkan (default).

Any other things to try ? i found one on forum that had the same problem - tried same version as stated there but still same issue.

**Comment by sum2012** [comment]
Have you tried older version ?

**Comment by vittrup88** [comment]
> Have you tried older version ?

I have tested v 1.10.3 from website official website download and the following:
ppsspp-v1.10.3-1646-g74f3542af-android.apk
ppsspp-v1.9.3-861-gfdd0b372a-android.apk

How old should I try ? any suggestion and I will test :)

**Comment by sum2012** [comment]
v1.7.4
https://www.ppsspp.org/files/1_7_4/ppsspp.apk

**Comment by vittrup88** [comment]
> v1.7.4
> https://www.ppsspp.org/files/1_7_4/ppsspp.apk

Tested and same error.
Did delete whole PSP folder and tested again and now working (no game tested)

So did install latest version with same method - and that is working now.
Will test with game now.

**Comment by hrydgard** [comment]
hm, you didn't happen to keep your old /PSP/SYSTEM/PPSSPP.ini around? that might have helped figuring this out...

**Comment by vittrup88** [comment]
> hm, you didn't happen to keep your old /PSP/SYSTEM/PPSSPP.ini around? that might have helped figuring this out...

No forgot that - I think the reason for this issue might have been problem overwriting the ini file. I have tried to recreate the problem without success.

I will start testing games to see if everything works.

**Comment by strulstrul** [comment]
I have the same issue with ppsspp on Android TV. Installed the emulator for the first time yesterday using first the version from store and later latest apk from site, 1.10.3. Also tried older versions. But only gets a black screen every time I start the app. When I press back on the remote or ps4 control the emulator interface shows for a blink before it quits. Never got the emulator to run. Tried delete the psp folder and reinstalled but same problem

My setup:
Philips TV 58PUS9005/12
Android ver 9

**Comment by vittrup88** [comment]
> I have the same issue with ppsspp on Android TV. Installed the emulator for the first time yesterday using first the version from store and later latest apk from site, 1.10.3. Also tried older versions. But only gets a black screen every time I start the app. When I press back on the remote or ps4 control the emulator interface shows for a blink before it quits. Never got the emulator to run. Tried delete the psp folder and reinstalled but same problem
> 
> My setup:
> Philips TV 58PUS9005/12
> Android ver 9

Have you edited the ppsspp.ini ?

**Comment by strulstrul** [comment]
No, only fresh installations. Is it the change of backend to OpenGL as described above you refering to?

**Comment by vittrup88** [comment]
> No, only fresh installations. Is it the change of backend to OpenGL as described above you refering to?

Yes. Set it to 0. And double check you got the file overwritten. That was my problem. The file ended in a download folder and not in the right place.

**Comment by strulstrul** [comment]
Worked like a charm. Big thanks.

**Comment by unknownbrackets** [comment]
Okay, so in both your cases you're on Android 8.1 or higher, where we default to Vulkan.  However, using Vulkan results in an entirely black screen, even in the UI (that is, before you load any game.)  Switching to OpenGL makes it work.

I thought we had almost the same issue on OpenGL, and it was related to compositing.  We ended up with a backbuffer that was getting blended and its alpha was zero, or something?  I might be remembering wrong.  We had to change the backbuffer parameters we were willing to choose.

Perhaps we're now seeing a similar issue with swapchain images?

-[Unknown]

**Comment by Gamemulatorer** [comment]
How about this now after https://github.com/hrydgard/ppsspp/commit/37b78fd40e6178da4ba9f0da44ef5818d29d1e4d

**Comment by TheErstwhile** [comment]
Not sure if this is helpful. On Jan 3rd 2023, I installed RetroArch from Ubuntu Application Store (SNAP). Added PPSSPP using the RetroArch Core Downloads. Had this problem, but it went away when I changed the Video Driver to ‘gl’ instead of ‘GLCore’. Has worked like a champ since.

**Comment by jimakujima** [comment]
> Not sure if this is helpful. On Jan 3rd 2023, I installed RetroArch from Ubuntu Application Store (SNAP). Added PPSSPP using the RetroArch Core Downloads. Had this problem, but it went away when I changed the Video Driver to ‘gl’ instead of ‘GLCore’. Has worked like a champ since.

I think i have to add this here, following this great hint from TheErstwhile. I'm using arch linux with nvidia 340xx dkms driver. Tried installing PPSSPP from the community repo and aur, both failed (starting the app with blank black screen with navigation sound), tried installing it through the flatpak also have the same issue.

Installed snapd, installed RetroArch using snap-store, download "Sony Playstation Portable (PPSSPP)" core. In my case, the "gl" video driver selected  by default. PSP games up and running pretty well, with stuttering/lagging sound (but that is another issue).

Thanks for the hint, @TheErstwhile. Cheers.

**Comment by JeffersonBledsoe** [comment]
I noticed this same issue on macOS (12.7.6). PPSSPP 1.17.1. Using OpenGL the app works fine, switching to Vulkan causes it to get stuck at a black screen on launch.

**Comment by hrydgard** [comment]
> I noticed this same issue on macOS (12.7.6). PPSSPP 1.17.1. Using OpenGL the app works fine, switching to Vulkan causes it to get stuck at a black screen on launch.

@JeffersonBledsoe can you use the menu (on the top of the screen) to switch back, or it is completely hung?

What specs are your mac? Primarily, which GPU?

I only test PPSSPP on x86 with Intel GPU, and Macbook Air M1 (ARM), since that's all the Macs I have.

**Comment by hrydgard** [comment]
Ok, yeah, it's known that Macs with Radeon have issues with our use of MoltenVK somehow. So stick to OpenGL for now, glad to hear the menu workaround works at least!
